### ISSUE TITLE
Replace UISearchDisplayController with UISearchController

### ISSUE THREAD

**Issue by tciuro** [comyarzaheri/Sol#28]
Hello,

Now that Xcode 7 is out, it would be great to see `Sol` updated for iOS 8 or iOS 9. The `UISearchDisplayController` class been deprecated in iOS 8 with `UISearchController`, so when running `Sol` in iOS8  or later, the app doesn't work well (it takes several attempts to search and select the city to see the weather).

**Comment by tciuro** [comment]
Hello Comyar, I'm curious as to why you're not using Storyboard. Is it a matter of preference? Thanks for the app! :-)

**Comment by comyarzaheri** [comment]
Yeah, really just preference. I never saw the advantage of using nibs and storyboards were really wonky when they first came out (i.e. impossible-to-resolve merge conflicts), so I just ended up writing my layout code by hand. Nowadays storyboards are much nicer, but old habits never die.

**Comment by Samzhang029** [comment]
Comer, would you like to fix this problem as I was working on iOS8.4? 
Sol works no well to display weather info. 
I tried to change the change UISearchDisplayController to UISearchController and met other errors. :(

**Comment by comyarzaheri** [comment]
I'm currently writing a Swift version of the app (with an updated/cooler UI!) which is why I haven't gotten around to fixing this.

**Comment by Samzhang029** [comment]
I want to use this SOL into my automation testing demo. Now everything is OK except the display the weather. 
I am not good at the Objective-C, so cannot fix the problems. 
Do you have the github link of your Swift version of the app? If it can work, I can change to this app.

**Comment by comyarzaheri** [comment]
I don't have the Swift version on GitHub yet...are you saying you're having trouble getting location search results working? I'm not clear on what the issue is here.

**Comment by Samzhang029** [comment]
Thanks for quick reply! 
No, app can search the locations.  
The problem is after selected location it cannot display the weather information and it indicated an unhappy face and "Update Failed. Check your network connection." 
I am sure we have the internet access. 
Maybe it's not the problem of UISearchDisplayController. Do you like to have a try?

**Comment by comyarzaheri** [comment]
Yeah that's definitely not going to happen because of the UISearchDisplayController. Do you have a valid Wunderground API key?

**Comment by Samzhang029** [comment]
Yes, we have a valid API_KEY. And if we use the web browser we can get the JSon. 
It's strange problem.

**Comment by comyarzaheri** [comment]
Hmm I'm not sure then...it should just work unless you're trying to add features, in which case it's likely just a bug on your end. I'd be happy to look at a code snippet or two if you think you can narrow down the problem to a few problem areas.

**Comment by niarilorn** [comment]
Not bad at all
